components/script/dom/htmlbodyelement.rs

```diff
@@ -47,6 +47,16 @@ impl HTMLBodyElement {
         let element = HTMLBodyElement::new_inherited(localName, prefix, document);
         Node::reflect_node(box element, document, HTMLBodyElementBinding::Wrap)
     }
+
+    /// https://drafts.csswg.org/cssom-view/#the-html-body-element
+    pub fn is_the_html_body_element(&self) -> bool {
+        let self_node = self.upcast::<Node>();
+        let root_elem = self.upcast::<Element>().get_root_element();
+        let root_node = root_elem.upcast::<Node>();
+        self_node.preceding_nodes(root_node).all(|n| !n.is::<HTMLBodyElement>()) &&
```

Comment:
`preceding_siblings`, and swap around the two conditions, so the cheap/most-likely to fail check goes first.